### PR TITLE
Dev 1.1.15 webank streamis fink load yaml

### PR DESCRIPTION
**Pull request by yangwenzea**
### What is the purpose of the change

Support param 'env.java.opts' in default flink config

### Brief change log

- Refactor linkis-engingplugin-flink to support param 'env.java.opts' 

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala [hunk 1]
```diff
@@ -232,6 +242,96 @@ class FlinkEngineConnFactory extends MultiExecutorEngineConnFactory with Logging
     context
   }
 
+  protected def getExtractJavaOpts(envJavaOpts: String): String = {
```
Comment (casionone) [inline linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala:245]: 调整为private 并添加单测 
单测使用org.junit.jupiter.api.Assertions.assertEquals 来断言判断 
单测目录因该和src/main 同级 即 src/test

Comment (casionone) [reply linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala:245]: 补充必要时代码注解

File: linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala [hunk 2]
```diff
@@ -232,6 +242,96 @@ class FlinkEngineConnFactory extends MultiExecutorEngineConnFactory with Logging
...
+      }
+    } else {
+      val inputStream = getClass.getResourceAsStream(yamlFile)
+      if (inputStream != null) {
+        val source = Source.fromInputStream(inputStream)
+        try {
+          val yamlContent = source.mkString
+          val yaml = new Yaml()
+          val configMap = yaml.loadAs(yamlContent, classOf[util.LinkedHashMap[String, Object]])
+          if (configMap.containsKey("env.java.opts")) {
```
Comment (casionone) [inline linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala:269]: env.java.opts  多个地方使用 定义为一个全局常量

File: linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala [hunk 3]
```diff
@@ -232,6 +242,96 @@ class FlinkEngineConnFactory extends MultiExecutorEngineConnFactory with Logging
...
+      } else {
+        throw new FileNotFoundException("YAML file not found in both file system and classpath.")
+      }
+    }
+    val merged = mergeAndDeduplicate(defaultJavaOpts, envJavaOpts)
+    merged
+  }
+
+
+  protected def mergeAndDeduplicate(str1: String, str2: String): String = {
```
Comment (casionone) [inline linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala:284]: 补充必要的注解 必要的地方可以加示例   
字符命名优化 str1 和str2  这种不好理解

File: linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+class TestFlinkEngineConnFactory {
+
+  @Test
+  def testMergeAndDeduplicate: Unit = {
+    var defaultJavaOpts = "-Da=3 -Db=4 -XXc=5 -Dk=a1=b";
+    var envJavaOpts = "-DjobName=0607_1 -Dlog4j.configuration=./log4j.properties -Da=1 -Dk=a1=c";
+    val merged = mergeAndDeduplicate(defaultJavaOpts, envJavaOpts)
+  }
+
+  protected def mergeAndDeduplicate(str1: String, str2: String): String = {
```
Comment (casionone) [inline linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala:31]: 不要重复定义 考虑修改为private

File: linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+import org.junit.jupiter.api.Test
+
+class TestFlinkEngineConnFactory {
+
+  @Test
+  def testMergeAndDeduplicate: Unit = {
+    var defaultJavaOpts = "-Da=3 -Db=4 -XXc=5 -Dk=a1=b";
+    var envJavaOpts = "-DjobName=0607_1 -Dlog4j.configuration=./log4j.properties -Da=1 -Dk=a1=c";
+    val merged = mergeAndDeduplicate(defaultJavaOpts, envJavaOpts)
```
Comment (casionone) [inline linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala:28]: 结果需要 断言判断

File: linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala (hunk 1 above)
Comment (yangwenzea) [reply linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala:245]: done

File: linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala (hunk 2 above)
Comment (yangwenzea) [reply linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala]: done

File: linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala (hunk 3 above)
Comment (yangwenzea) [reply linkis-engineconn-plugins/flink/src/main/scala/org/apache/linkis/engineconnplugin/flink/factory/FlinkEngineConnFactory.scala]: done

File: linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala (hunk 4 above)
Comment (yangwenzea) [reply linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala]: done

File: linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala (hunk 5 above)
Comment (yangwenzea) [reply linkis-engineconn-plugins/flink/src/main/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala]: done

File: linkis-engineconn-plugins/flink/src/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    val yamlFilePath = FLINK_CONF_DIR.getValue
+    val yamlFile = yamlFilePath + "/" + FLINK_CONF_YAML.getHotValue()
+    if (new File(yamlFile).exists()) {
+      val source = Source.fromFile(yamlFile)
+      try {
+        val yamlContent = source.mkString
+        val yaml = new Yaml()
+        val configMap = yaml.loadAs(yamlContent, classOf[util.LinkedHashMap[String, Object]])
+        if (configMap.containsKey("env.java.opts")) {
+          defaultJavaOpts = configMap.get("env.java.opts").toString
```
Comment (casionone) [inline linkis-engineconn-plugins/flink/src/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala:44]: 这里也使用  FLINK_ENV_JAVA_OPTS

Comment (yangwenzea) [reply linkis-engineconn-plugins/flink/src/test/scala/org.apache.linkis.engineconnplugin.flink/factory/TestFlinkEngineConnFactory.scala]: done

**Review by casionone — APPROVED** [review]

**Pull request merged**